components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/definitions/OAS3Parser.java

```diff
@@ -257,82 +246,102 @@ private String getJsonExample(Schema model, Map<String, Schema> definitions) {
      */
     private String getXmlExample(Schema model, Map<String, Schema> definitions) {
         Example example = ExampleBuilder.fromSchema(model, definitions);
-        return new XmlExampleSerializer().serialize(example);
+        String rawXmlExample = new XmlExampleSerializer().serialize(example);
+        return rawXmlExample.replace("<?xml version='1.1' encoding='UTF-8'?>", "");
     }
 
     /**
-     *Sets default script
+     *Sets default script for response codes without defined payloads
      *
      * @param genCode String builder
      */
-    private void setDefaultGeneratedResponse(StringBuilder genCode) {
-        genCode.append("/* mc.setProperty('CONTENT_TYPE', 'application/json');\n\t" +
-                "mc.setPayloadJSON('{ \"data\" : \"sample JSON\"}');*/\n" +
-                "/*Uncomment the above comment block to send a sample response.*/");
+    private void setDefaultGeneratedResponse(StringBuilder genCode, String responseCode) {
+        genCode.append("\n/*if (!responses[").append(responseCode).append("]) {\n").append("  responses[")
+                .append(responseCode).append("] = [];\n").append("}\n").append("responses[")
+                .append(responseCode).append("][\"application/(json or xml)\"] = {}/<>*/\n");
     }
 
     /**
-     * Generates string for variables in Payload Generation
+     * Generates string for initializing response code arrays and payload variables
      *
      * @param responseCode response Entry Code
      * @param example generated Example Json/Xml
      * @param type  mediaType (Json/Xml)
+     * @param initialized response code array
      * @return generatedString
      */
-    private String getGeneratedResponseVar(String responseCode, String example, String type) {
-        return "\nvar response" + responseCode + type + " = " + example + "\n\n";
-    }
-
-    /**
-     * Generates string for methods in Payload Generation
-     *
-     * @param responseCode response Entry Code
-     * @param type mediaType (Json/Xml)
-     * @return manualCode
-     */
-    private String getGeneratedSetResponse(String responseCode, String type) {
-        return "  mc.setProperty('HTTP_SC', \"" + responseCode + "\");\n" +
-                "  mc.setProperty('CONTENT_TYPE', 'application/" + type + "');\n" +
-                "  mc.setPayload" + type.toUpperCase() + "(response" + responseCode + type + ");";
+    private String getGenRespPayloads(String responseCode, String example, String type, boolean initialized) {
```

Comment:
Please use full names. getGeneratedResponsPayloads or the one that you meant to say.